ruby/red-arrow/lib/arrow/csv-loader.rb

```diff
@@ -93,10 +93,18 @@ def reader_options
       @options.each do |key, value|
         case key
         when :headers
-          if value
-            options.n_header_rows = 1
+          case value
+          when ::Array
+            options.column_names = value
+          when ::String
+            return nil
+          when false, nil
+            options.generate_column_names = true
+          when :first_line, true
+            options.generate_column_names = false
           else
-            options.n_header_rows = 0
+            # Undocumented in Ruby CSV, but it treats truthy as true
+            options.generate_column_names = false
```

Comment:
We can simplify this:

```ruby
case value
when ::Array
  options.column_names = value
when String # Preceding :: is needless because we don't have Arrow::String
  return nil
else
  if value
    options.generate_column_names = false
  else
    options.generate_column_names = true
  end
end
```